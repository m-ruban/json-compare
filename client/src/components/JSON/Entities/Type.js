import React from 'react';
import PropTypes from 'prop-types';
import { Badge } from 'reactstrap';

function getType(value) {
    let type = typeof value;

    if (type === 'object') {
        if (Array.isArray(value)) {
            type = 'array';
        }

        if (value === null) {
            type = 'null';
        }
    }

    return type;
}

const Type = ({ value }) => {
    const type = getType(value);
    return <Badge color={type}>{type}</Badge>;
};

Type.propTypes = {
    value: PropTypes.any,
};

export default Type;
