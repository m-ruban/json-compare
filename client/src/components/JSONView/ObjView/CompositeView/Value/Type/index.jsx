import React from 'react';
import PropTypes from 'prop-types';
import { getType } from 'modules/compare';
import Mark from 'components/Mark';

const Type = ({ value }) => {
    const type = getType(value);
    return <Mark type={type} text={type} />;
};

Type.propTypes = {
    value: PropTypes.any,
};

export default Type;
