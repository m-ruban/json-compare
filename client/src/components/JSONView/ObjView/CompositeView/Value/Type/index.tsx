import React from 'react';

import Mark from 'components/Mark';
import { getType } from 'modules/compare';

const Type = ({ value }: { value: string; }) => {
    const type: string = getType(value);
    return <Mark type={type} text={type} />;
};

export default Type;
