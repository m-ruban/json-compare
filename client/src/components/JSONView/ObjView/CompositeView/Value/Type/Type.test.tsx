import React from 'react';

import Type from 'components/JSONView/ObjView/CompositeView/Value/Type';
import { render } from 'enzyme';

describe('<Key />', () => {
    it('Renders correctly', () => {
        const props = { value: 'test' };
        const output = render(<Type {...props} />);
        expect(output).toMatchSnapshot();
    });
});
