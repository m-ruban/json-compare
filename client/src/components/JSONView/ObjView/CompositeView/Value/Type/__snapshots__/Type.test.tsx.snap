// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Key /> Renders correctly 1`] = `
<span
  class="json-mark json-mark_string badge badge-json"
>
  string
</span>
`;
